src/app/components/views/ticket-generator/ticket-generator.component.ts

```diff
@@ -23,26 +23,26 @@ export class TicketGeneratorComponent implements OnInit {
   public trainingDue!: string[]; // Array of training titles
 
   constructor() {}
-  
+
   public ngOnInit(): void {
     this.banForm = new FormGroup({
       'ban': new FormControl(null, [
-        Validators.required, 
-        Validators.minLength(15), 
-        Validators.maxLength(15), 
+        Validators.required,
+        Validators.minLength(15),
+        Validators.maxLength(15),
         Validators.pattern(/^[0-9]+$/m) // Only allow numbers
       ]),
     });
 
     this.trainingForm = new FormGroup({
       'uuid': new FormControl(null, [
-        Validators.required, 
-        Validators.minLength(7), 
-        Validators.maxLength(7), 
+        Validators.required,
+        Validators.minLength(7),
+        Validators.maxLength(7),
         Validators.pattern(/[a-z]+[a-z]+[0-9]+[0-9]+[0-9]+[a-z0-9]+[0-9]/) // only allow 2 letters, 3 numbers, and then a letter OR number, and then finally another number
```

Comment:
Ok I was going to criticize the regex here but then I realized I think I wrote it

Reply:
you did. thanks buddy.

Reply:
`/[a-z]{2}\d{3}[a-z\d]\d/i` I think will do the same as above. The `i` at the end will also make it case insensitive, if we want that